Depth fix

mjallais opened the pull request.

Commit:
COMP: Remove depth variable from the XML file to avoid confusion

Commit:
ENH: Set spacing in the output image

thewtex commented on app/AcquireIntersonArrayBMode.cxx:
```diff
@@ -147,13 +147,18 @@ int main( int argc, char * argv[] )
 
+  const double ns = ContainerType::MAX_SAMPLES; // number of samples
+  const double fs = frequencies[ frequencyIndex ] / 1000; // [kHz]=[samples/ms] - frequency per sample
+  double depth = c * ns / ( 2 * fs );
```
const

thewtex commented on app/AcquireIntersonArrayBMode.cxx:
```diff
@@ -147,13 +147,18 @@ int main( int argc, char * argv[] )
   const int scanWidth = container.GetWidthScan();
   const int scanHeight = container.GetHeightScan();
   std::cout << "Width = " << width << std::endl;
   std::cout << "Height = " << height << std::endl;
```
It would be more precise to rename `Width = ` to `Axial samples = ` and `Height = ` to `Lateral samples =`

aylward replied:
I think we should follow the notation used in the IntersonArray documentation.

thewtex commented on app/AcquireIntersonArrayBMode.cxx:
```diff
@@ -147,13 +147,18 @@ int main( int argc, char * argv[] )
   std::cout << "Height = " << height << std::endl;
   std::cout << "ScanWidth = " << scanWidth << std::endl;
   std::cout << "ScanHeight = " << scanHeight << std::endl;
   std::cout << "MM per Pixel = " << container.GetMmPerPixel() << std::endl;
```
I think these should be removed, because they are scan converter related, which is not relevant to our output.

aylward replied:
Are we accessing the bmode data prior to scan conversion?   My impression was that bmode data we received was post scan conversion.    In the driver for the old interson probe the scan converter was separate.   For this probe the scan converter seems to be a requirement.   HardInitScanConverter must be called to get any data from the probe.

thewtex replied:
Ah, makes sense.

Since (I assume) the RF output is not using the scan converter, then these changes should be made in `AcquireIntersonArrayRF` and not in `AcquireIntersonArrayBMode`.

So `imageSize[0]` is always a fixed 512, or does this change with the the depth setting? Is `container.MAX_SAMPLES` unused?

We should use `container.GetMmPerPixel()` for `imageSpacing[0]`? And for `imageSpacing[1]`, or is that fixed?

thewtex commented on app/AcquireIntersonArrayBMode.cxx:
```diff
@@ -147,13 +147,18 @@ int main( int argc, char * argv[] )
     }
 
+  const double ns = ContainerType::MAX_SAMPLES; // number of samples
+  const double fs = frequencies[ frequencyIndex ] / 1000; // [kHz]=[samples/ms] - frequency per sample
```
`frequencies` is the excitation center frequency, which is different from the sampling frequency. The sampling frequency is a fixed value for this probe.

thewtex commented on app/AcquireIntersonArrayBMode.xml:
```diff
@@ -59,5 +59,12 @@
       <maximum>255</maximum>
     </integer>
+    <double>
+      <name>c</name>
```
We could change this from `c` to `sos` -- single character variables are difficult to search for in code.

thewtex commented on app/AcquireIntersonArrayRF.xml:
```diff
@@ -49,5 +49,12 @@
       <maximum>100</maximum>
     </integer>
+    <double>
+      <name>c</name>
```
-> `sos`

thewtex reviewed (changes requested):
Very nice, thank you!

A few comments inline.

aylward commented on app/AcquireIntersonArrayRF.cxx:
```diff
@@ -127,12 +127,18 @@ int main( int argc, char * argv[] )
+  container.IdleInitScanConverter( depth_acq, width_samples, height_lines, probeId,
       steering, false, false, 0 );
-  container.HardInitScanConverter( depth, width_samples, height_lines, steering );
+  container.HardInitScanConverter( depth_acq, width_samples, height_lines, steering );
```
I am still not certain this is right.  They have a "mmperpixel" function, and the value returned by that function changes when depth changes, and the bmode image that is returned changes when this depth value changes.

thewtex commented on app/AcquireIntersonArrayRF.cxx:
```diff
@@ -146,6 +152,11 @@ int main( int argc, char * argv[] )
   image->SetRegions( imageRegion );
+  ImageType::SpacingType imageSpacing;
+  imageSpacing[ 0 ] = c / ( 2 * fs );
+  imageSpacing[ 1 ] = 38.0 / height_lines;
```
`38.0 / height_lines` -> `38.0 / (height_lines - 1)`

thewtex commented on app/AcquireIntersonArrayRF.cxx:
```diff
@@ -127,12 +127,18 @@ int main( int argc, char * argv[] )
 
+  const double ns = ContainerType::MAX_RFSAMPLES; // number of samples
+  const double fs = frequencies[ frequencyIndex ] / 1000; // [kHz]=[samples/ms] - frequency per sample
+  double depth = c * ns / ( 2 * fs );
```
`ns` -> `ns - 1`

samuelgerber committed:
Fix Buffer Size

The buffer size in the Bmode image was to large. The buffer size is unaffacted
by scanWidth and scanHeight in the ScanConverter and appers to have a fixed size
of NBOLINES x MAX_SAMPLES.

Now teh BMode images is received at full depth and the image size does not ahve to
be set to MAX_SMAPLES/2.

samuelgerber committed:
Updated AquireBmode app to corret buffer size

samuelgerber committed:
Cleanup

thewtex commented on app/AcquireIntersonArrayBMode.cxx:
```diff
@@ -146,13 +147,18 @@ int main( int argc, char * argv[] )
 
+  const double ns = ContainerType::MAX_SAMPLES; // number of samples
+  const double fs = 17000; // [kHz]=[samples/ms] - sampling frequency
+  const double depth = sos * ns / ( 2 * fs );
```
Concluding conversations offline, the B-mode uses the setting of the internal scan converter, so we do not need these values in `AcquireIntersonArrayBMode.cxx`.

thewtex commented on app/AcquireIntersonArrayBMode.cxx:
```diff
@@ -168,6 +174,11 @@ int main( int argc, char * argv[] )
   imageRegion.SetSize( imageSize );
   image->SetRegions( imageRegion );
+  ImageType::SpacingType imageSpacing;
+  imageSpacing[ 0 ] = sos / ( 2 * fs );
```
Here we should use `container.GetMmPerPixel()` as I understand it.

thewtex commented on app/AcquireIntersonArrayBMode.cxx:
```diff
@@ -168,6 +174,11 @@ int main( int argc, char * argv[] )
   image->SetRegions( imageRegion );
+  ImageType::SpacingType imageSpacing;
+  imageSpacing[ 0 ] = sos / ( 2 * fs );
+  imageSpacing[ 1 ] = 38.0 / height;
```
`38.0 / (height - 1)`

thewtex commented on app/AcquireIntersonArrayBMode.xml:
```diff
@@ -59,15 +59,12 @@
-      <minimum>0</minimum>
-      <maximum>200</maximum>
-    </integer>
+    <double>
```
Since we are using their scan converter, we do not control the assumed speed of sound in `AcquireIntersonArrayBMode`, so this parameter can be removed.

thewtex commented on app/AcquireIntersonArrayRF.cxx:
```diff
@@ -127,12 +127,18 @@ int main( int argc, char * argv[] )
   const itk::SizeValueType framesToCollect = frames;
 
+  const double ns = ContainerType::MAX_RFSAMPLES; // number of samples
+  const double fs = 17000; // [kHz]=[samples/ms] - sampling frequency
```
This should be `30000` according to specifications.

thewtex reviewed (changes requested):
Looking great.  Added comments to make improvements per our discussion. Looking forward to having the right aspect ratio on these image :-)

aylward committed:
Merge branch 'master' into DepthFix
